core/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -204,10 +204,13 @@ public String getLocalFilename(int blockIndex) throws IOException {
    * @throws IOException
    */
   public List<String> getLocationHosts() throws IOException {
+    List<String> ret = new ArrayList<String>();
```

Comment:
Makes sense. Got your point. 

Then the default behavior should be 
1. create an empty list
2. add location one by one
3. return back list (if nothing added, it is empty;)